src/raftstore/store/peer.rs

```diff
@@ -484,6 +494,43 @@ impl Peer {
         StaleState::Valid
     }
 
+    fn calculate_lease_expired_time(&self,
+                                    send_to_quorum_ts: Timespec,
+                                    quorum_commit_ts: Timespec)
+                                    -> Option<Timespec> {
+        // The valid leader lease should be
+        // "lease = election_timeout - (quorum_commit_ts - send_to_quorum_ts)"
+        // And the expired timestamp for that leader lease is "quorum_commit_ts + lease".
```

Comment:
If it takes a longer time then election_timeout to commit a log, it will rollback the `expired_time`. This should not happen since it's obvious not the lease time needed for this case.